### DISCUSSION TITLE
USB Floppy Drive

### DISCUSSION THREAD

**Discussion by paradoxnj** [hathach/tinyusb#2860] [Q&A]
I am trying to read data from a 1.44MB USB Floppy Drive (Part number 19308801-19).  TinyUSB seems to recognize it as an MSC but it does not mount it as a storage controller.  What I mean by that is tuh_msc_mount_cb() is never called but tuh_mount_cb() is.  Would anyone know how to manually mount it as an MSC?  According to OSDev Wiki, I should be able to issue it SCSI commands to read and write from it so it should be compatible.

https://wiki.osdev.org/Floppy_Disk_Controller

"All [USB](https://wiki.osdev.org/USB) devices, including USB floppy drives, are accessed indirectly (using SCSI-style commands encoded in USB datapackets) over the USB bus. USB floppy drives do not use any of the IO ports or FDC commands described in this article."

**Comment by JustAnother1** [comment]
Have you tried to access a thumb drive? Does that work?

**Comment by paradoxnj** [comment]
Thumb drives work fine.  They mount perfectly.  USB floppies do not.  For giggles, I tried a USB CD ROM and it goes into mount/unmount loop.Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Lars Pötter ***@***.***> Date: 10/30/24  3:40 PM  (GMT-05:00) To: hathach/tinyusb ***@***.***> Cc: Anthony Rufrano ***@***.***>, Author ***@***.***> Subject: Re: [hathach/tinyusb] USB Floppy Drive (Discussion #2860) 
Have you tried to access a thumb drive? Does that work?

**Comment by hathach** [comment]
1.4" may use different protocol UFI/CBI instead of supported BOT (thumbdrive). Which reader you are testing with, I don't have one but will try to see if I could get one easy enough.

**Comment by paradoxnj** [comment]
1.44MB USB Floppy Drive (Part number 19308801-19)

https://esaitech.com/products/teac-27l4226-19308801-19-fru-05k9283-external-1-44mb-floppy-drive

**Reply by hathach** [reply]
I order an 1.44mb usb reader to test with, it is not exact the same but is probably good enough. It may take 2 weeks for it to deliver. I will give it a try when I could manage the time

**Reply by paradoxnj** [reply]
Hi.  I am just following up to see if you received your USB Floppy yet.  Thanks.

**Reply by hathach** [reply]
yeah, it just got delivered, though I forgot to order an floopy disk to test with. But I am currently busy anyway, so please be patient.

**Reply by paradoxnj** [reply]
Just following up.  I would really like to use TinyUSB for this.

**Reply by hathach** [reply]
Unfortunately, I still haven't got time to look at this yet.

**Reply by paradoxnj** [reply]
Any luck?

**Reply by hathach** [reply]
unfortunately, atm I need time more than luck which is still lacking.
